x-pack/legacy/plugins/lens/server/usage/collectors.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { CallCluster } from 'src/legacy/core_plugins/elasticsearch';
+import { CoreSetup, SavedObjectsLegacyService } from 'src/core/server';
+import { getVisualizationCounts } from './visualization_counts';
+import { LensUsage } from './types';
+
+export function getSavedObjectsClient(
+  savedObjects: SavedObjectsLegacyService,
+  callAsInternalUser: unknown
+) {
+  const { SavedObjectsClient, getSavedObjectsRepository } = savedObjects;
+  const internalRepository = getSavedObjectsRepository(callAsInternalUser);
+  return new SavedObjectsClient(internalRepository);
+}
+
+export function registerLensUsageCollector(
+  core: CoreSetup,
+  plugins: {
+    savedObjects: SavedObjectsLegacyService;
+    usage: {
+      collectorSet: {
+        makeUsageCollector: (options: unknown) => unknown;
+        register: (options: unknown) => unknown;
+      };
+    };
+  }
+) {
+  const lensUsageCollector = plugins.usage.collectorSet.makeUsageCollector({
+    type: 'lens',
+    fetch: async (callCluster: CallCluster): Promise<LensUsage> => {
+      const savedObjectsClient = getSavedObjectsClient(plugins.savedObjects, callCluster);
+      try {
+        return getVisualizationCounts(savedObjectsClient);
+      } catch (err) {
+        return {
+          saved_total: 0,
```

Comment:
Is this the right behavior? It seems like it'd skew our data if errors are happening. Not sure what else we should do, and I know nothing about the usage collector system, so this may be the right thing. It just feels wrong, and makes me doubt our metrics, if this is the recommended approach.